cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -405,6 +416,47 @@ private File saveHConf(Configuration hConf, File targetDir) throws IOException {
   }
 
   /**
+   * Localizes resources requested by users in the Spark Program's client (or beforeSubmit) phases.
+   * In Local mode, also copies resources to a temporary directory.
+   *
+   * @param contextConfig the {@link SparkContextConfig} for this Spark program
+   * @param clientContext the {@link ClientSparkContext} for this Spark program
+   * @param allLocalizedResources the list of all (user-requested + CDAP system) {@link LocalizeResource} to be
+   *                                 localized for this Spark program
+   * @param targetDir in local mode, a temporary directory to copy the resources to
+   * @return a {@link Map} of resource name to the {@link File} handle for the local file.
+   */
+  private Map<String, File> localizeUserResources(SparkContextConfig contextConfig,
+                                                  ClientSparkContext clientContext,
+                                                  List<LocalizeResource> allLocalizedResources,
+                                                  File targetDir) throws IOException {
+    Map<String, File> localizedResources = new HashMap<>();
+    Map<String, LocalizeResource> resourcesToLocalize = clientContext.getResourcesToLocalize();
+    for (final Map.Entry<String, LocalizeResource> entry : resourcesToLocalize.entrySet()) {
+      final File localizedFile;
+      if (contextConfig.isLocal()) {
+        // in local mode, also localize resources in a temporary directory
+        File tempFile = new File(targetDir, entry.getKey());
```

Comment:
You should refactor this local file copying mechanism to be shared with MapReduce as well.